Move Python egg-info to top level directory

tiran opened the pull request:
All setup.py use the same build, dist and *.egg-info directory on top
level. Build artefacts are no longer placed in local build directories.

Signed-off-by: Christian Heimes cheimes@redhat.com

tiran commented:
The PR will also simplifies @pspacek next branch a bit.

tiran commented:
I ran into some issues with ```make rpm```. ```setup.py``` started to pick up build artefacts of other packages. Now every package uses its own subdirectory under the top-level ```build``` directory.

pspacek commented:
Why this change? It goes against the usual behavior where build artifacts are placed in the same directory as sources. Also, I intend to support ouf-of-source-tree builds (aka [VPATH](https://www.gnu.org/software/automake/manual/html_node/VPATH-Builds.html)) and I'm not sure it will work with your patch.

I would reject this change if there is no strong need.

stlaz commented:
+1 with @pspacek, build artefacts should be in the same directory as is their source. I would like to have them removed on `make clean` if that does not currently work.

tiran commented:
I have changed the PR a bit

* dist and build are no longer moved
* as discussed on IRC ```freeipa.egg-info``` is now ```ipaserver.egg-info```. pkg_resource assumes that a egg-info directory has the same prefix as a package directory.
* I added a egg_info command plugin based on https://blog.kevin-brown.com/programming/2014/09/24/combining-autotools-and-setuptools.html . My variant uses the distutils API properly and corrects a bug with ```install_egg_info``` plugin.
* ipasetup now calculates the package root path based on the path of setup.py. This should be enough to enable VPATH out-of-tree builds.
* By default the egg-info directories are in the top-level project dir or on top_builddir. For in-tree tests the egg-info directories must be in sys.path in order to register entry points correctly, see PR #187

tiran commented:
I'm closing this PR because the issue can't be fixed w/o patching setuptools heavily. I got it mostly right except for SOURCES.txt. We need to revise the project structure at a later point.
